build.sbt

```diff
@@ -292,6 +292,10 @@ lazy val scioTest: Project = Project(
 ).settings(
   commonSettings ++ itSettings,
   description := "Scio helpers for ScalaTest",
+  // necessary to properly test since we need this value at compile time
+  initialize in Test ~= { _ =>
+    System.setProperty( "OVERRIDE_TYPE_PROVIDER", "com.spotify.scio.bigquery.validation.SampleOverrideTypeProvider" )
```

Comment:
There could be only 1 type provider per project right? How does it work if one wants to override multiple types?

Reply:
currently only 1 type provider per project in the future we could extend it to provide a list of type providers and just take the first one that matches

if you want to override multiple types you have to contain it in the same type provider

Reply:
I guess the easiest would be just to support a comma separated list of class names in the JVM property?

Reply:
yeah we should probably just do that now

but in general having one provider to provide multiple types is the paradigm we're going to use internally

Reply:
decided to leave this as is since we can extend in the future to allow for multiple types